### PR TITLE
Paged atlas support for GUI

### LINKED ISSUE #7998: Paged atlases for gui
It wil be nice to have paged atlasses in gui, to reuse images, from go space.

now we can't use paged atlases, because many images common for gui and go space

### PR DESCRIPTION
**Pull request by Jhonnyg**
GUI components can now use paged atlases and materials.

Fixes #7998 

## PR checklist

* [x] Code
	* [x] Add engine and/or editor unit tests.
	* [x] New and changed code follows the overall code style of existing code
	* [x] Add comments where needed
* [ ] Documentation
	* [x] Make sure that API documentation is updated in code comments
	* [ ] Make sure that manuals are updated (in github.com/defold/doc)
* [ ] Prepare pull request and affected issue for automatic release notes generator
	* [x] Pull request - Write a message that explains what this pull request does. What was the problem? How was it solved? What are the changes to APIs or the new APIs introduced? This message will be used in the generated release notes. Make sure it is well written and understandable for a user of Defold.
	* [x] Pull request - Write a pull request title that in a sentence summarises what the pull request does. Do not include "Issue-1234 ..." in the title. This text will be used in the generated release notes.
	* [x] Pull request - Link the pull request to the issue(s) it is closing. Use on of the [approved closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).
	* [x] Affected issue - Assign the issue to a project. Do not assign the pull request to a project if there is an issue which the pull request closes.
	* [ ] Affected issue - Assign the "breaking change" label to the issue if introducing a breaking change.
	* [ ] Affected issue - Assign the "skip release notes" is the issue should not be included in the generated release notes.

### CODE REVIEW

File: com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/AtlasBuilder.java [hunk 1]
```diff
@@ -74,41 +101,91 @@ public Task<TextureImage.Type> create(IResource input) throws IOException, Compi
...
-        }
+        boolean compress = project.option("texture-compression", "false").equals("true");
+        TextureImage texture = generateTextureImage(result.images, textureImageType, texProfile, compress);
 
-        TextureImage texture = TextureUtil.createCombinedTextureImage(textureImages, textureType);
         task.output(0).setContent(textureSet.toByteArray());
         task.output(1).setContent(texture.toByteArray());
     }
+
+    public static void main(String[] args) throws IOException, CompileExceptionError {
```
Comment (Jhonnyg) [inline com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/AtlasBuilder.java:141]: We use this to build test content now as well

File: engine/gamesys/src/waf_gamesys.py [hunk 2]
```diff
@@ -65,14 +65,9 @@ def transform_texture_name(task, name):
 def transform_tilesource_name(name):
     name = name.replace('.tileset', '.t.texturesetc')
     name = name.replace('.tilesource', '.t.texturesetc')
+    name = name.replace('.atlas', '.t.texturesetc')
     return name
 
-def transform_textureset_filename(task, name):
```
Comment (Jhonnyg) [inline engine/gamesys/src/waf_gamesys.py:70]: not used

File: editor/src/clj/editor/gl/shader.clj [hunk 3]
```diff
@@ -507,6 +507,23 @@ of GLSL strings and returns an object that satisfies GlBind and GlEnable."
...
+      (str "The Material expects a paged Atlas, but the selected " image-property-name " is not paged")
+
+      (and (not is-paged-material)
+           (pos? texture-page-count))
+      (str "The Material does not support paged Atlases, but the selected " image-property-name " is paged")
+
+      (< material-max-page-count texture-page-count)
+      (str "The Material's 'Max Page Count' is not sufficient for the number of pages in the selected " image-property-name))))
+
+(def page-count-mismatch-error-message (memoize page-count-mismatch-error-message-raw))
```
Comment (vlaaad) [inline editor/src/clj/editor/gl/shader.clj:525]: Not sure why `memoize` is needed here, the `page-count-mismatch-error-message-raw` fn looks very simple and light.

**Review by vlaaad — APPROVED** [review]
LGTM!

**Review by JCash — APPROVED** [review]
LGTM!

File: editor/src/clj/editor/gl/shader.clj (hunk 3 above)
Comment (matgis) [reply editor/src/clj/editor/gl/shader.clj:525]: I was concerned about allocating a lot of identical error strings while building. Theoretically you could have a lot of these errors after editing the material or atlas from every sprite and gui objects that make use of it.

**Pull request merged**